### PR TITLE
feat(feel): Clarify edge cases in FEEL expressions

### PR DESCRIPTION
**Pull request by saig0**
## Description

Extend the FEEL documentation to clarify edge cases, related to the following issues:
- https://github.com/camunda/feel-scala/pull/887
- https://github.com/camunda/feel-scala/issues/801
- https://github.com/camunda/feel-scala/issues/731
- https://github.com/camunda/feel-scala/issues/867  

The changes are relevant for all versions 8.2+.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by saig0** [comment]
Changed pages:
- https://preview.docs.camunda.cloud/pr-4222/docs/next/components/modeler/feel/language-guide/feel-unary-tests/
- https://preview.docs.camunda.cloud/pr-4222/docs/next/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion/#numberfrom
- https://preview.docs.camunda.cloud/pr-4222/docs/next/components/modeler/feel/language-guide/feel-temporal-expressions/#literal
- https://preview.docs.camunda.cloud/pr-4222/docs/next/components/modeler/feel/builtin-functions/feel-built-in-functions-string/#substringstring-start-position-length

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31"` which is invalid because June has
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date, for example, `2024,6,31` which is invalid because
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31T10:00:00"` which is invalid because
```

File: docs/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (nicpuppa) [inline docs/components/modeler/feel/language-guide/feel-temporal-expressions.md:44]:
```suggestion
`@"2024-06-31"` which is invalid because June has only 30 days.
```

File: docs/components/modeler/feel/language-guide/feel-unary-tests.md
Comment (nicpuppa) [inline docs/components/modeler/feel/language-guide/feel-unary-tests.md:132]:
```suggestion
When a unary operator is not enough to express the condition, any expression that returns a boolean value can be used,
```

File: versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31"` which is invalid because June has
```

File: versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date, for example, `2024,6,31` which is invalid because
```

File: versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.2/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31T10:00:00"` which is invalid because
```

File: versioned_docs/version-8.2/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (nicpuppa) [inline versioned_docs/version-8.2/components/modeler/feel/language-guide/feel-temporal-expressions.md:44]:
```suggestion
`@"2024-06-31"` which is invalid because June has only 30 days.
```

File: versioned_docs/version-8.2/components/modeler/feel/language-guide/feel-unary-tests.md
Comment (nicpuppa) [inline versioned_docs/version-8.2/components/modeler/feel/language-guide/feel-unary-tests.md:131]:
```suggestion
When a unary operator is not enough to express the condition, any expression that returns a boolean value can be used,
```

File: versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31"` which is invalid because June has
```

File: versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date, for example, `2024,6,31` which is invalid because
```

File: versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.3/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31T10:00:00"` which is invalid because
```

File: versioned_docs/version-8.3/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (nicpuppa) [inline versioned_docs/version-8.3/components/modeler/feel/language-guide/feel-temporal-expressions.md:44]:
```suggestion
`@"2024-06-31"` which is invalid because June has only 30 days.
```

File: versioned_docs/version-8.3/components/modeler/feel/language-guide/feel-unary-tests.md
Comment (nicpuppa) [inline versioned_docs/version-8.3/components/modeler/feel/language-guide/feel-unary-tests.md:132]:
```suggestion
When a unary operator is not enough to express the condition, any expression that returns a boolean value can be used,
```

File: versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31"` which is invalid because June has
```

File: versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date, for example, `2024,6,31` which is invalid because
```

File: versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.4/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31T10:00:00"` which is invalid because
```

File: versioned_docs/version-8.4/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (nicpuppa) [inline versioned_docs/version-8.4/components/modeler/feel/language-guide/feel-temporal-expressions.md:44]:
```suggestion
`@"2024-06-31"` which is invalid because June has only 30 days.
```

File: versioned_docs/version-8.4/components/modeler/feel/language-guide/feel-unary-tests.md
Comment (nicpuppa) [inline versioned_docs/version-8.4/components/modeler/feel/language-guide/feel-unary-tests.md:132]:
```suggestion
When a unary operator is not enough to express the condition, any expression that returns a boolean value can be used,
```

File: versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31"` which is invalid because June has
```

File: versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date, for example, `2024,6,31` which is invalid because
```

File: versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (nicpuppa) [inline versioned_docs/version-8.5/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date, for example, `"2024-06-31T10:00:00"` which is invalid because
```

File: versioned_docs/version-8.5/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (nicpuppa) [inline versioned_docs/version-8.5/components/modeler/feel/language-guide/feel-temporal-expressions.md:44]:
```suggestion
`@"2024-06-31"` which is invalid because June has only 30 days.
```

File: versioned_docs/version-8.5/components/modeler/feel/language-guide/feel-unary-tests.md
Comment (nicpuppa) [inline versioned_docs/version-8.5/components/modeler/feel/language-guide/feel-unary-tests.md:132]:
```suggestion
When a unary operator is not enough to express the condition, any expression that returns a boolean value can be used,
```

**Review by nicpuppa — COMMENTED** [review]
Hey @saig0 I left some small wording suggestions please have a look 👀

**Review by nicpuppa — DISMISSED** [review]
Thanks @saig0 🚀

**Comment by mesellings** [comment]
@saig0 I will review this for you 👍

File: docs/components/modeler/feel/language-guide/feel-temporal-expressions.md
Comment (mesellings) [inline docs/components/modeler/feel/language-guide/feel-temporal-expressions.md:48]:
```suggestion
Adds a value to another value. The operator is defined for the following types.
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-string.md
Comment (mesellings) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-string.md:33]:
```suggestion
Returns a substring of the given value, starting at `start position` with the given `length`. If `length` is greater than
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (mesellings) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:77]:
```suggestion
Returns `null` if the string is not a valid calendar date. For example, `"2024-06-31"` is invalid because June has
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (mesellings) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:108]:
```suggestion
Returns `null` if the components don't represent a valid calendar date. For example, `2024,6,31` is invalid because
```

File: docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md
Comment (mesellings) [inline docs/components/modeler/feel/builtin-functions/feel-built-in-functions-conversion.md:190]:
```suggestion
Returns `null` if the string is not a valid calendar date. For example, `"2024-06-31T10:00:00"` is invalid because
```

**Review by mesellings — CHANGES_REQUESTED** [review]
Lgtm @saig0 - I've left a few changes/suggestions that need to be applied to the other versions as well, and a single typo, but happy to approve if you accept these?

**Review by mesellings — APPROVED** [review]
Lgtm 👍

**Comment by saig0** [comment]
@mesellings thank you for your fast feedback and suggestions. :+1:

**Comment by saig0** [comment]
@mesellings / @nicpuppa I need another approval. I rebased the branch to `main`.

**Review by nicpuppa — APPROVED** [review]

**Pull request merged**
